### DISCUSSION TITLE
Could OPA store the policy local storage automatically by this: https://www.openpolicyagent.org/docs/latest/rest-api/#create-or-update-a-policy

### DISCUSSION THREAD

**Discussion by PaulYuanJ** [open-policy-agent/community#410] [Community]
I'm using docker contanier to start OPA as server, I want to get the policy from the host after I put a example policy into opa by the manual https://www.openpolicyagent.org/docs/latest/rest-api/#create-or-update-a-policy

But I can't find the policy anywhere on the host.

I use below command to start the OPA Server
`docker run --rm --privileged -d -p 8181:8181 \
-v /data/server/opa_rules:/policies --name=opa2 openpolicyagent/opa:0.35.0 \
run --server /policies --log-format=json-pretty --set=decision_logs.console=true`

**Comment by charlieegan3** [comment]
Hi Paul, the policies updated in the REST api like this are not persisted to disk. You might want to look into using the bundle service API instead to allow OPA to load policies over the network: https://www.openpolicyagent.org/docs/latest/management-bundles/

Bundles have a setting `persist: true` which means they are stored on disk if you need to inspect them. Hope that helps!

**Reply by PaulYuanJ** [reply]
Thanks for you quick response. :)
And I have put a xx.repo into /policies dir, but it could not be loaded before restart the container. If it is possible to hot-load these policies?

**Reply by charlieegan3** [reply]
Watching like this isn't something that's currently supported but it might land soon in: https://github.com/open-policy-agent/opa/pull/5812

However, if you need to reload policy at all like this I'd really suggest using the bundle API for that job instead.
